### DISCUSSION TITLE
Getting Local Key

### DISCUSSION THREAD

**Discussion by Sivivatu** [make-all/tuya-local#766] [Q&A] [answered]
Im following the process for collecting local keys described [here](https://github.com/make-all/tuya-local#finding-your-device-id-and-local-key). 

However, in the Tuya Dev portal, everytime I try to test the Get the Device information endpoint, every response is 
```
{
  "code": 1106,
  "msg": "permission deny",
  "success": false,
  "t": 1685991014456,
  "tid": "ce9b64b903d111eea4908a3120fb3e2a"
}
```

Has Tuya changed the process and broken this or am I doing something wrong?

Cheers

**Comment by make-all** [comment, marked as answer]
There are many similarly named functions in the API, maybe two or three accessible ones return the local keys, and only one returns dp_ids in case your device is not yet supported.  The error you see comes from some of the functions that are intended for managing devices by a large commercial operator, and can also occur on all functions if you have not selected the server/region correctly.

**Reply by Sivivatu** [reply]
>There are many similarly named functions in the API

This is exactly right. I was checking the "Get the device information" endpoint, which fails but checking "Get Device Information" was successful.

Thanks for the guide
